stdlib/REPL/src/LineEdit.jl

```diff
@@ -1238,6 +1243,48 @@ _edit_indent(buf::IOBuffer, b::Int, num::Int) =
     num >= 0 ? edit_splice!(buf, b => b, ' '^num, rigid_mark=false) :
                edit_splice!(buf, b => (b - num))
 
+# return the identifier under the cursor, possibly with other words concatenated
+# to it with dots (e.g. "A.B.C" in "X; A.B.C*3", if the cursor is between "A" and "C")
+function current_word_with_dots(buf::IOBuffer)
+    pos = position(buf)
+    while true
+        char_move_word_right(buf)
+        if eof(buf) || peek(buf, Char) != '.'
+            break
+        end
+    end
+    pend = position(buf)
+    while true
+        char_move_word_left(buf)
+        p = position(buf)
+        p == 0 && break
+        seek(buf, p-1)
+        if peek(buf, Char) != '.'
+            seek(buf, p)
+            break
+        end
+    end
+    pbegin = position(buf)
+    word = pend > pbegin ?
+        String(buf.data[pbegin+1:pend]) :
+        ""
+    seek(buf, pos)
+    word
+end
+
+current_word_with_dots(s::MIState) = current_word_with_dots(buffer(s))
+
+function activate_module(s::MIState)
+    word = current_word_with_dots(s);
+    isempty(word) && return beep(s)
+    try
+        REPL.activate(Base.Core.eval(Base.active_module(),
+                                     Base.Meta.parse(word)))
```

Comment:
TODO: add a note that `Core.eval` is used, because `Mod.eval(e)` would work fine in general but not when `Mod` is `Core`.

Reply:
`mod.eval` is also (nearly) always wrong, so I'm not sure it makes sense to comment about that in particular

Reply:
Ok, will not add a comment :) But I thought that `mod.eval` is nearly always correct, except for bare modules (hence incorrrect in the general case).